1-Draft/RFCXXXX-ScriptBlockLogging.md

```diff
@@ -0,0 +1,213 @@
+---
+RFC: RFCXXXX
+Author: Travis Plunk
+Status: Draft
+SupersededBy:
+Version: 1.0
+Area: Engine
+Comments Due: 08/15/2019
+Plan to implement: Yes
+---
+
+# Title
+
+ScriptBlock Logging
+
+## Description
+
+This is an addition to the existing ScriptBlock creation logging.
+The goal is to make it easier to configure to collect the logs centrally.
+This feature will be experimental.
+
+## Motivation
+
+As a administrator of machines which run PowerShell (machine admins),
+I can easily log stript blocks which PowerShell executes to systems such as Azure Log Analytics or Splunk (log systems),
+so that I can analyze PowerShell activity on all the machines I run.
+
+## User Experience
+
+This example assumes you have registered the key to communicate
+with Azure Log Analytics with PowerShell Secrets with the name `myKeyName`.
+
+**Note:** `myKeyName` is a secret register using the system added in the [Secrets Management RFC][secrets-rfc]
+
+### First time registration
+
+```powershell
+Register-PSScriptBlockLogger -Provider AzureLogAnalytics -KeyName myKeyName -Properties @{
+    WorkspaceId = ‘myworkspaceid’
+}
+```
+
+```output
+Provider: AzureLogAnalytics
+Properties: {WorkspaceId="myworkspaceid"}
+KeyName: myKeyName
+```
+
+### Register when something is already registered
+
+```powershell
+Register-PSScriptBlockLogger -Provider AzureLogAnalytics -KeyName myKeyName -Properties @{
+    WorkspaceId = ‘myworkspaceid’
+}
+```
+
+```output
+Error: A provider is already registered.  User -Force to overwrite the current provider settings.
+```
+
+### Unregister
+
+```powershell
+Unregister-PSScriptBlockLogger
+}
+```
+
+There would be no output.
+
+### Get current registration
+
+Note: this example assumes that the provider was already register as in [First time registration](#first-time-registration)
+
+```powershell
+Get-PSScriptBlockLogger
+```
+
+```output
+Provider: AzureLogAnalytics
+Properties: {WorkspaceId="myworkspaceid"}
+KeyName: myKeyName
+```
+
+## Specification
+
+### Assumptions
+
+1. The transport to the Logging system is secure.
+   Secure includes:
+   - Encrypted
+   - Authenticated
+1. The machine admin is willing and authorized to send all ScriptBlocks to the logging system.
+
+### Registration
+
+Only one logging provider can be registered at a time.
+There will be two options for registration,
+which are explained below:
+
+#### Registration cmdlet
+
+There will be a cmdlet to register a logging provider.
+`Register-PSScriptBlockLogger` will allow you to register a provider.
```

Comment:
`Provider` can be confusing as I think most people think of PowerShell Provider.  Perhaps use the word extension.  I don't see where you define the model for extensions.  Perhaps also show the user experience if 3rd parties publish extensions to PSGallery and how they get registered.